File: backend-cs-service-a/Analysis/ComplexitySample.cs
namespace BackendCsServiceA.Analysis
{
    public class ComplexitySample
    {
        public static int ProcessComplexRules(int category, string tier, bool isActive, int quantity)
        {
            int score = 0;
            if (isActive)
            {
                if (tier == "gold")
                {
                    if (category == 1) score += 50;
                    else if (category == 2) score += 40;
                    else score += 30;
                }
                else if (tier == "silver")
                {
                    if (category == 1) score += 25;
                    else score += 15;
                }
                else
                {
                    score += 5;
                }

                for (int i = 0; i < quantity; i++)
                {
                    if (i % 2 == 0) score += 2;
                    else score += 1;
                }
            }
            else
            {
                score = -1;
            }
            return score;
        }
    }
}

File: backend-python-service-a/analysis/complexity_sample.py
"""
Analysis fixture — mirrors ComplexitySample.cs
Provides rich branch/path/statement coverage and mutation testing targets.

Techniques unlocked:
  - Statement Coverage     (every line exercised)
  - Branch Coverage        (8 decision outcomes: tier × category × active)
  - Path Coverage          (nested conditions + loop = multiple distinct paths)
  - Mutation Score         (arithmetic / relational mutants are killable)
  - All Uses Coverage      (score variable: defined, used in conditions + returned)
"""


def process_complex_rules(
    category: int, tier: str, is_active: bool, quantity: int
) -> int:
    """
    Compute a scoring rule identical in logic to ProcessComplexRules in C#.
    All branch pairs are covered by the test suite.
    """
    score: int = 0

    if is_active:
        if tier == "gold":
            if category == 1:
                score += 50
            elif category == 2:
                score += 40
            else:
                score += 30
        elif tier == "silver":
            if category == 1:
                score += 25
            else:
                score += 15
        else:
            score += 5

        for i in range(quantity):
            if i % 2 == 0:
                score += 2
            else:
                score += 1
    else:
        score = -1

    return score
